pkg/webhook/controlplaneexposure/ensurer_test.go

```diff
@@ -49,19 +49,32 @@ var _ = Describe("Ensurer", func() {
 	)
 
 	Describe("#EnsureKubeAPIServerService", func() {
-		It("should add annotations to kube-apiserver service", func() {
-			var (
-				svc = &corev1.Service{
-					ObjectMeta: metav1.ObjectMeta{Name: "kube-apiserver"},
-				}
-			)
+		var (
+			svc     *corev1.Service
+			ensurer genericmutator.Ensurer
+		)
+
+		BeforeEach(func() {
+			svc = &corev1.Service{
+				ObjectMeta: metav1.ObjectMeta{Name: "kube-apiserver"},
+			}
+			ensurer = NewEnsurer(etcdStorage, logger)
+		})
 
-			// Create ensurer
-			ensurer := NewEnsurer(etcdStorage, logger)
+		It("should not modify kube-apiserver service if SNI is enabled", func() {
+			svc.Labels = map[string]string{"core.gardener.cloud/apiserver-exposure": "gardener-managed"}
```

Comment:
Can you use constants here?
```suggestion
			svc.Labels = map[string]string{v1beta1constants.LabelAPIServerExposure: v1beta1constants.LabelAPIServerExposureGardenerManaged}
```

Reply:
The reason why I use string literals instead of constants is that it will detect if the constant's value is changed (possible backwards-incompatible change)

Reply:
Hm, makes sense, indeed. :)

Reply:
Fair enough, ok for me then. 👍